templates/common/infra/bicep/core/database/cosmos/mongo/cosmos-mongo-cluster.bicep

```diff
@@ -0,0 +1,62 @@
+metadata description = 'Azure Cosmos DB MongoDB vCore cluster'
+@maxLength(40)
+param name string
+param location string = resourceGroup().location
+param tags object = {}
+
+param administratorLogin string
+param sku string
+param storage int
+param nodeCount int
+@secure()
+param administratorLoginPassword string
+param highAvailabilityMode bool = false
+param allowAzureIPsFirewall bool = false
+param allowAllIPsFirewall bool = false
+param allowedSingleIPs array = []
+
+resource mognoCluster 'Microsoft.DocumentDB/mongoClusters@2023-03-01-preview' = {
+  name: name
+  tags: tags
+  location: location
+  properties: {
+    administratorLogin: administratorLogin
+    administratorLoginPassword: administratorLoginPassword
+    nodeGroupSpecs: [
+        {
+            diskSizeGB: storage
+            enableHa: highAvailabilityMode
+            kind: 'Shard'

```

Comment:
Should this be a param

Reply:
Currently there is no other option..
Only Shard available from the docs here https://learn.microsoft.com/en-us/azure/templates/microsoft.documentdb/mongoclusters?pivots=deployment-language-bicep

Reply:
on it